### PR TITLE
Added .travis.yml and appveyor.yml for automatic release

### LINKED ISSUE #75: Add Travis and Appveyor intergration
CI should do the following
- [ ] Lint Testing
- [ ] Generate Binaries

### PR DESCRIPTION
**Pull request by akshitkrnagpal**
Resolves #75.

To setup
- Install Appveyor and Travis.
- Add "GH_TOKEN" env variable in Travis and Appveyor.

To release artifacts
- Draft a release with tag "v" + version number from package.json (like v0.1.0)
- Push commits. It will publish all artifacts automatically.
- Release when ready.

PS - Merging this commit will not publish artifacts. After merging, well have to draft a new release, push commit which changes the version in `package.json`. And the release will get published.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+image: Visual Studio 2017
+
+platform:
+  - x64
```
Comment (saghul) [inline appveyor.yml]: Can we also do 32bit builds on Windows? On macOS and Linux I´m ok with just providing 64bit builds, but there are many old Windows machines out there...

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - libx11-dev
+            - zlib1g-dev
+            - libpng12-dev
+            - libxtst-dev
+            - g++-4.8
```
Comment (saghul) [inline .travis.yml]: What Ubuntu version is this, Trusty? Can you try to use the Xenial image? It will have more modern versions of these. Not sure how this affects anything though.

File: package.json [hunk 3]
```diff
@@ -24,18 +24,19 @@
       "!main.js"
     ],
     "mac": {
-      "artifactName": "${productName}-${version}-${arch}.${ext}",
+      "artifactName": "jitsi-meet-${version}-${arch}.${ext}",
```
Comment (saghul) [inline package.json]: Why is this needed? Isn't that variable properly used?

**Comment by saghul** [comment]
> Add "GH_TOKEN" env variable in Travis and Appveyor.

What should this token contain?

Also, have you tested this on your own fork? If not, can you please do it so I can see how it looks like? Thanks!

File: appveyor.yml (hunk 1 above)
Comment (akshitkrnagpal) [reply appveyor.yml]: This is the version of the os image to be used. (Because we can build for both on 64-bit)
It is currently building one `.exe` file for both 32-bit and 64-bit version.

**Comment by akshitkrnagpal** [comment]
> What should this token contain?

It has to be a Github Personal Access Token. This will be used to upload artifacts.

> Also, have you tested this on your own fork?

You can see https://github.com/akshitkrnagpal/jitsi-meet-electron/releases

I used my `ci-test` branch for this release. https://github.com/akshitkrnagpal/jitsi-meet-electron/tree/ci-test

**Comment by akshitkrnagpal** [comment]
Also, For signing `dmg` we'll have to set environment variable according to this page. https://www.electron.build/code-signing

Currently if skips signing for mac os because environment variable is not found.

File: appveyor.yml (hunk 1 above)
Comment (saghul) [reply appveyor.yml]: Excellent, thanks!

File: .travis.yml (hunk 2 above)
Comment (saghul) [reply .travis.yml]: Thanks!

File: package.json (hunk 3 above)
Comment (saghul) [reply package.json]: gotcha. can we drop the arch part from the macOS name? it's going to be x64 always anyway

**Comment by saghul** [comment]
> You can see https://github.com/akshitkrnagpal/jitsi-meet-electron/releases

Looking good! Can the unnecessary assets be dropped? Also, can we build 32bit appimages? I believe it was possible using docker

**Comment by saghul** [comment]
> Also, For signing dmg we'll have to set environment variable according to this page. https://www.electron.build/code-signing
> 
> Currently if skips signing for mac os because environment variable is not found.

I see. It also looks like a developer ID is necessary for signing, even for distribution outside of the store, sigh. let's keep the signing out of this PR and I'll work on that when I have some time.

File: package.json (hunk 3 above)
Comment (akshitkrnagpal) [reply package.json]: Yeah. I can remove that. But It is actually automatically being dropped. I guess we don't need to do that.

**Comment by akshitkrnagpal** [comment]
> Can the unnecessary assets be dropped?

They aren't unnecessary actually. 
Blockmap files are used for partial updates.

> Also, can we build 32bit appimages? I believe it was possible using docker

I'll try that.

> let's keep the signing out of this PR 

Actually, there will be no need for changing anything in here. Just setting the environment variable should be enough.

**Comment by saghul** [comment]
> They aren't unnecessary actually.
> Blockmap files are used for partial updates.

Today I learned :-)



> Actually, there will be no need for changing anything in here. Just setting the environment variable should be enough.

Excellent!

**Comment by saghul** [comment]
@akshitkrnagpal One last question: I get this error when building on Linux x64:

```
• rebuilding native production dependencies platform=linux arch=ia32
Error: /usr/bin/node exited with code 1
Output:

> jitsi-meet-electron-utils@0.5.2 install /home/saghul/src/jitsi-meet-electron/node_modules/jitsi-meet-electron-utils
> prebuild-install || node-gyp rebuild

make: Entering directory '/home/saghul/src/jitsi-meet-electron/node_modules/jitsi-meet-electron-utils/build'
  CXX(target) Release/obj.target/sourceId2Coordinates/node_addons/sourceId2Coordinates/src/index.o
make: Leaving directory '/home/saghul/src/jitsi-meet-electron/node_modules/jitsi-meet-electron-utils/build'

Error output:
prebuild-install WARN install No prebuilt binaries found (target=2.0.3 runtime=electron arch=ia32 platform=linux)
cc1plus: warning: command line option ‘-Wbad-function-cast’ is valid for C/ObjC but not for C++
In file included from /usr/include/features.h:452,
                 from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdint.h:26,
                 from /usr/lib/gcc/x86_64-redhat-linux/8/include/stdint.h:9,
                 from /home/saghul/.electron-gyp/iojs-2.0.3/deps/v8/include/v8.h:19,
                 from /home/saghul/.electron-gyp/iojs-2.0.3/src/node.h:63,
                 from ../node_addons/sourceId2Coordinates/src/index.cc:1:
/usr/include/gnu/stubs.h:7:11: fatal error: gnu/stubs-32.h: No such file or directory
 # include <gnu/stubs-32.h>
           ^~~~~~~~~~~~~~~~
compilation terminated.
make: *** [sourceId2Coordinates.target.mk:104: Release/obj.target/sourceId2Coordinates/node_addons/sourceId2Coordinates/src/index.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules.bundled/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 4.17.3-200.fc28.x86_64
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules.bundled/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/saghul/src/jitsi-meet-electron/node_modules/jitsi-meet-electron-utils
gyp ERR! node -v v8.11.3
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jitsi-meet-electron-utils@0.5.2 install: `prebuild-install || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the jitsi-meet-electron-utils@0.5.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/saghul/.npm/_logs/2018-07-13T05_45_45_226Z-debug.log

    at ChildProcess.childProcess.once.code (/home/saghul/src/jitsi-meet-electron/node_modules/builder-util/src/util.ts:251:14)
    at Object.onceWrapper (events.js:317:30)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
From previous event:
    at rebuild (/home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/out/util/yarn.js:238:18)
    at /home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/util/yarn.ts:20:11
    at Generator.next (<anonymous>)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
From previous event:
    at installOrRebuild (/home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/out/util/yarn.js:68:17)
    at /home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/packager.ts:450:13
    at Generator.next (<anonymous>)
From previous event:
    at Packager.installAppDependencies (/home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/packager.ts:411:70)
    at /home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/packager.ts:361:20
    at Generator.next (<anonymous>)
From previous event:
    at Packager.doBuild (/home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/packager.ts:337:39)
    at /home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/packager.ts:313:52
    at Generator.next (<anonymous>)
    at /home/saghul/src/jitsi-meet-electron/node_modules/graceful-fs/graceful-fs.js:99:16
    at /home/saghul/src/jitsi-meet-electron/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqWrap.oncomplete (fs.js:135:15)
From previous event:
    at Packager._build (/home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/packager.ts:290:133)
    at /home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/packager.ts:286:23
    at Generator.next (<anonymous>)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
From previous event:
    at Packager.build (/home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/packager.ts:243:14)
    at /home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/src/index.ts:52:40
    at Generator.next (<anonymous>)
From previous event:
    at build (/home/saghul/src/jitsi-meet-electron/node_modules/electron-builder-lib/out/index.js:220:17)
    at build (/home/saghul/src/jitsi-meet-electron/node_modules/electron-builder/src/builder.ts:224:10)
    at then (/home/saghul/src/jitsi-meet-electron/node_modules/electron-builder/src/cli/cli.ts:46:19)
    at <anonymous>
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jitsi-meet-electron@1.0.0-alpha.1 dist: `npm run build && electron-builder`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the jitsi-meet-electron@1.0.0-alpha.1 dist script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/saghul/.npm/_logs/2018-07-13T05_45_45_282Z-debug.log

```
Should I remove ia32 from package.json ?

**Comment by saghul** [comment]
Nevermind, I went for it :-)
